### ISSUE TITLE
Multiplication error with secp256k1.ModNScalar

### ISSUE THREAD

**Issue by MicOestergaard** [decred/dcrd#2777]
It seems that for some special values multiplication gives different results depending on the order of the scalars. Here is a code snippet that demonstrates this behaviour:

```
r, _ := new(big.Int).SetString("fffffffffffffffffffffffffffffffebaaedce6af48a03bbfd25e8cd0364140", 16)
s, _ := new(big.Int).SetString("55555555555555555555555555555554e8e4f44ce51835693ff0ca2ef01215c2", 16)

var r2, s2, res1, res2 secp256k1.ModNScalar
r2.SetByteSlice(r.Bytes())
s2.SetByteSlice(s.Bytes())

expected := new(big.Int).Mul(r, s)
expected.Mod(expected, secp256k1.Params().N)
fmt.Println("Expected:", hex.EncodeToString(expected.Bytes()))

res1.Mul2(&r2, &s2)
fmt.Println("r*s:     ", res1.String())

res2.Mul2(&s2, &r2)
fmt.Println("s*r:     ", res2.String())

if !res1.Equals(&res2) {
    panic("not equal")
}
```

**Comment by davecgh** [comment]
Thanks for the report.  I'll take a look shortly.

**Comment by davecgh** [comment]
This is resolved by #2778.  I will tag a patch release for the  module once that is merged.

**Comment by davecgh** [comment]
This is fixed in patch v4.0.1 of the `dcrd/dcrec/secp256k1/v4` module.
